decision/engine.go

```diff
@@ -289,7 +292,10 @@ func (e *Engine) addBlock(block blocks.Block) {
 	for _, l := range e.ledgerMap {
 		l.lk.Lock()
 		if entry, ok := l.WantListContains(block.Cid()); ok {
-			e.peerRequestQueue.Push(l.Partner, entry)
+			e.peerRequestQueue.PushBlock(l.Partner, peertask.Task{
+				Identifier: entry.Cid,
```

Comment:
I'm a _bit_ worried about allocations here. This used to be a bit of a hot-spot until we made entries and CIDs by-value.

This is a great abstraction so I'm not going to block on that but we just need to keep it in mind.

Reply:
Yea I mean, it's still technically by-value -- just a new value that makes a new allocation :) I will keep an eye on it as this goes into production.